### ISSUE TITLE
Form Validator non-required fields should still run validation

### ISSUE THREAD

**Issue by Stetzon** [wffranco/vue-strap#76]
My use case is where I display a "User profile update" form where the user can change their email, name, etc., but they do not have to re-enter their password when submitting the form.  Therefore, I am removing the **required** attribute from the password inputs so the form validation will pass and the user can submit the form.  With the **required** attribute missing, much of the form validation falls through if the user does decide to update their password and fill out the fields.

Example [fiddle](https://jsfiddle.net/50wL7mdz/20305/)

1. Password field bypasses the minlength check if we start to define a password.

2. Password match field does not run validation until we enter text in that box.

In short, I'm thinking that required attribute should only be checked if the fields are empty, otherwise we still need to run the other validation checks
